content/en/blog/_posts/2024-06-20-pod-failure-policy-ga

```diff
@@ -0,0 +1,124 @@
+---
+layout: blog
+title: "Kubernetes 1.31: Pod failure policy goes GA"
+date: 2024-06-20
+slug: kubernetes-1-31-pod-failure-policy-stable
+author: >
+  Michał Woźniak (Google)
+---
+
+This blog discusses the Pod Failure Policy
+[Pod failure policy](/docs/concepts/workloads/controllers/job/#pod-failure-policy)
+for Jobs as it graduates to stable in 1.31.
+
+## Why to use it? {#pod-failure-policy-why-to-use}
+
+Consider running a large computational workload, comprising thousands of
+long-running pods executing on thousands of Nodes. In that case you need
+to account for Pod failures so that your workload can continue running till
+successful completion.
+
+Prior to Pod failure policy, the only built-in mechanism in Kubernetes to
+account for Pod failures was the based on the `.backoffLimit` field, which
+lets you tolerate the specified number of Pod failures. However, this mechanism
+is not flexible enough.
```

Comment:
we also have `BackoffLimitPerIndex`. Can we consider that?

Reply:
It could help to some extent, in some scenarios, but its main objective is different: allow to execute all indexes despite failures. So I think for the most part it is orthogonal.  I think there is no point in complicating the introduction. I could mention it quickly somewhere if you have a good idea how.

Reply:
In the last commit I include an example for backoff limit per index. PTAL.